python/src/main/java/org/apache/zeppelin/python/PythonCondaInterpreter.java

```diff
@@ -115,22 +115,33 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
   private void changePythonEnvironment(String envName)
       throws IOException, InterruptedException {
     PythonInterpreter python = getPythonInterpreter();
+    String condaLibraryPath = null;
     String binPath = null;
     if (envName == null) {
       binPath = getProperty(ZEPPELIN_PYTHON);
       if (binPath == null) {
         binPath = DEFAULT_ZEPPELIN_PYTHON;
       }
+      condaLibraryPath = null;
     } else {
       Map<String, String> envList = getCondaEnvs();
+      for (String key : envList.keySet()) {
+      }
       for (String name : envList.keySet()) {
         if (envName.equals(name)) {
           binPath = envList.get(name) + CONDA_PYTHON_PATH;
+          condaLibraryPath = String.format("%s/lib", envList.get(name));
           break;
         }
       }
     }
-    python.setPythonCommand(binPath);
+
+    File pythonBin = new File(binPath);
+    if (pythonBin.exists() && pythonBin.isFile() && pythonBin.canExecute()) {
+      python.setCurrentCondaEnvName(envName);
+      python.setPythonCondaLibPath(condaLibraryPath);
+      python.setPythonCommand(binPath);
```

Comment:
Is `setPythonCondaLibPath` necessary ? Is `setPythonCommand` enough ?

IMO, we could use `python.getProperty().setProperty('zeppelin.python', binPath)` .

Reply:
... for binPath
`binPath = python.getProperty().setProperty('zeppelin.python', binPath)`
and then go into this block with `pythonBin = new File(binPath);`

Reply:
The new conda python environment is not related to `zeppelin.python`.
(If you have a new installation of python using conda)
We can freely change the python environment in a new environment via conda.
Therefore, you must register an additional Python library path for the new environment for zeppelin py4j gateway.
Otherwise you will not be able to use the python library installed via conda.

In version 0.7, there is an add library operation.
but in the current version, it seems that the structure has been changed and it is missing.
I modified it.

Reply:
@cloverhearts I don't understand why `you must register an additional Python library path for the new environment for zeppelin py4j gateway`. This is not consistent with the behavior when I use conda in shell. e.g. I create an env using conda in mac terminal, then I just need to set `zeppelin.python` to use this conda env.

Reply:
`` `
python.getProperty (). setProperty ('zeppelin.python', binPath)
`` `

This is not the process for setting up Zeppelin Python.
It is the action that changes the environment of Python Interpreter by judging whether the user installs new Python through Conda.

There are two cases.

1. Does not install Python via Conda
User has installed another library via conda.
In this case, the Python environment should not be changed in the default environment of zeppelin.
I checked it out and in the worst case Zeppelin's Python Interpreter is not recovered because of an infinite loop.
It's a defense code for it.

2. Install Python via Conda.
If user has installed a Python environment via Conda,
When conda activate (env name), Zeppelin should change user Python environment to user's new settings.
And if you do a Conda deactivate, Zeppelin should return to the original Zeppelin setting.
Therefore, Zeppelin can not use the binPath setting as follows:
`` `
python.getProperty (). setProperty ('zeppelin.python', binPath)
`` `

Reply:
@zjffdu 
Thank you for your opinion,
I present scenarios for environment changes using conda.
Python version and module names are arbitrary, so do not worry.
What is important is change in every environment.

1. Basically Zeppelin's python
python: 2.71
installed library: (none)

use to zeppelin
```
%python
import sys
print(sys.version)
print(sys.path)
```
result
```
python version 2.7.1
%ZEPPELIN_HOME%/interpreter/python/lib:%PYTHON_HOME
```
```
%python
import myModule
```
result
```
no module myModule
```


2. install conda and new python 
```
%python.conda create --name python3 python=3.4
```
```
%python.conda activate python3
```
```
%python
import sys
print(sys.version)
print(sys.path)
```
result
```
python version 3.4
%ZEPPELIN_HOME%/interpreter/python/lib:%PYTHON_HOME:%CONDA_ENV%/python3/lib
```
```
%python
import myModule
```
result
```
no module myModule
```
try again
```
%python.conda install myModule
```

```
import myModule
```
result
```
imported.
```

3. restore python environment
```
%python.conda deactivate
```
check for recovered to the original Zeppelin environment.
```
%python
import sys
print(sys.version)
print(sys.path)
```
result
```
python version 2.7.1
%ZEPPELIN_HOME%/interpreter/python/lib:%PYTHON_HOME
```
```
import myModule
```
result
```
no module myModule
```

All three must be supported by conda.
1. Support to installing and using the python library
2. Support to change the python version
3. Support to configuring non-python environments

old zeppelin does works for this part(0.71).
but, the Python Interpreter has changed significantly since 0.71.
I guess missing the environment part for this.
I modified it.


python env set code
(https://github.com/apache/zeppelin/blob/master/python/src/main/java/org/apache/zeppelin/python/PythonInterpreter.java#L151)